app/models/screenshot.rb

```diff
@@ -108,7 +108,14 @@ def generate_thumbnail(content)
 
   def fetch
     Rails.logger.debug("Fetching screenshot from #{source_url}")
-    content = open(source_url, "rb", :read_timeout => 6)
+    begin
+      content = open(source_url, "rb", :read_timeout => 6)
+    rescue Exception => e
```

Comment:
We should only rescue the particular exception resulting from invalid redirect. Hopefully it will even contain the new url so we don't need to guess it.